service/compute/api.go

```diff
@@ -233,6 +233,114 @@ func (a *ClustersAPI) Impl() ClustersService {
 	return a.impl
 }
 
+// WaitGetClusterRunning repeatedly calls [ClustersAPI.Get] and waits to reach RUNNING state
+func (a *ClustersAPI) WaitGetClusterRunning(ctx context.Context, clusterId string,
+	timeout time.Duration, callback func(*ClusterInfo)) (*ClusterInfo, error) {
+	ctx = useragent.InContext(ctx, "sdk-feature", "long-running")
+	return retries.Poll[ClusterInfo](ctx, timeout, func() (*ClusterInfo, *retries.Err) {
+		clusterInfo, err := a.Get(ctx, GetClusterRequest{
+			ClusterId: clusterId,
+		})
+		if err != nil {
+			return nil, retries.Halt(err)
+		}
+		if callback != nil {
+			callback(clusterInfo)
+		}
+		status := clusterInfo.State
+		statusMessage := clusterInfo.StateMessage
+		switch status {
+		case StateRunning: // target state
+			return clusterInfo, nil
+		case StateError, StateTerminated:
+			err := fmt.Errorf("failed to reach %s, got %s: %s",
+				StateRunning, status, statusMessage)
+			return nil, retries.Halt(err)
+		default:
+			return nil, retries.Continues(statusMessage)
+		}
+	})
+}
+
+// WaitGetClusterRunning is a wrapper that calls [ClustersAPI.WaitGetClusterRunning] and waits to reach RUNNING state.
+type WaitGetClusterRunning[R any] struct {
+	Response *R
```

Comment:
If the suggestion is applied then this no longer needs a type param or this response field.